storage_service/locations/models/package.py

```diff
@@ -1931,10 +1932,19 @@ def delete_from_storage(self):
                 error = lom._delete_update_lom(self, delete_lom_ids)
         try:
             self.current_location.space.delete_path(self.full_path)
-        except Exception as e:
-            error = e.message
-
-        # Remove pointer file, and the UUID quad directories if they're empty
+        except (StorageException, NotImplementedError, ValueError) as err:
+            # The package hasn't been deleted and the deletion request needs
+            # to be resubmitted, so status is UPLOADED as it was prior to the
+            # deletion request. We also want to avoid Schrödinger's package
+            # when trying to delete something already DELETED. Only set status
+            # back to UPLOADED state if it hasn't already been set to DELETED.
+            # This will mean it is still visible in Archivematica.
+            if self.status != self.DELETED:
+                self.status = self.UPLOADED
```

Comment:
I agree with the reasoning in the comment here, but it seems like there may be some other status cases (VERIFIED, etc). We are already setting package status on success only in the view — maybe it would be simpler to just not update package status at all in this function (remove the update on line 1964).

Reply:
That sounds like a plan. I'll take a look. (I was torn between that and not removing `1964` the second opinion helps!)

Reply:
I have logged a ticket against this [here](https://github.com/archivematica/Issues/issues/760). We only have one state in memory at this point in the code's execution, `DEL_REQ` and you are right to say there are more states than `UPLOADED`. As such, I will implement a fix to make sure we set this back to the previous state. That will be `DEL_REQ`. I'm not a fan, but  1. I feel this is worth investigating further without changing too much of the service's behavior and 2. this change now feels out of scope for where this PR began as we reach code-freeze, and it would only be a good idea if it were a genuinely quick fix.

Reply:
Previous comment was incorrect. Will remove this here.